articles/your-first-exteranto.md

```diff
@@ -0,0 +1,295 @@
+
+```bash
+npm run watch
+```
+
+Warning of unresolved dependencies can be fixed with freash npm install.
+
+Once the script is active you should see the following output:
+
+```bash
+  DONE  Compiled successfully
+
+  ✔ Manifest file found.
+  ✔ [chrome] extension packed.
+  ✔ [safari] extension packed.
+  ✔ [extensions] extension packed.
+
+  Your extension was successfully built.
+```
+
+### Load the Unpacked Extension
+
+Running `watch` will execute a project build, resulting in `dist` and `packs`
+folders being created in the top level of the directory.
+
+`dist` contains the transpiled extension files `background.js` and `content.js`.
+
+`packs` contains uncompressed browser extension packages to be loaded in the
+browser locally. Loading from packs into the browser, we have the benefit of hot
+reloading during debugging and development, though the extension still needs to
+be manually refreashed from within the browser itself.
+
+To install the extension locally for develpment and debugging open Chrome and
+navigate to the `chrome://extensions` page where you can toggle developer mode,
+'load unpacked' extension and choose the 'chromeextension' folder from `packs`.
+[Browser Extension Basics][browser-extension-basics] has further details.
+
+You will see an entry for 'Browser Extension 0.0.1' listed (the extension name
+and version are specified in `manifest.json`). Click the link to the 'background
+page' and a console window will open diplaying the Exteranto welcome message.
+
+
+### Outline
+
+For this tutorial, we will create a very simple extension with no UI, just a
+background script that listens for a specific browser event and then dispatches
+a message to the content script, which will perform some action.
+
+1. Define a message to be dispatched between backgound and content
+2. Dfine and register the background script listener
+3. Define and register the content script listener
+
+The browser event that triggers our background script will be 'Browser Action'
+click, meaning that the toolbar icon associated to our installed extension has
+been clicked by the user.
+
+In order to register this event we must provide access in the project config.
+Navigate to to `config/index.ts` and under `providers` add an entry for
+`BrowserActionProvider`. We will need to ensure that the provider is imported by
+the config:
+
+```typescript
+  import { BrowserActionProvider } from '@exteranto/api'
+```
+
+### Create a Message
+
+In `src/messages` create a `PingContentScriptMessage.ts`, open and add the
+following:
+
+```typescript
+  import { Message } from '@exteranto/core'
+
+  export class PingContentScriptMessage extends Message {
+    //
+  }
+```
+
+The `Message` class itself extends `Event` and is defined in Exteranto core
+modules. We need not define any properties on the `PingContentScriptMessage`
+class, since the action of recieving it will be enough for our purposes here,
+but we could provide it a `payload` or a `respond` callback.
```

Comment:
respond is provided by default on every message. I'd say "but we could provide additional data in the constructor that can be accessed by the message receiver" or something similar.